### PR TITLE
Improve the documentation for adding `bridge-metadata.json`

### PR DESCRIPTION
**Pull request by iwahbe**
This is in response to https://github.com/pulumi/pulumi-terraform-bridge/issues/1690.

### CODE REVIEW

**Comment by iwahbe** [comment]
CC @ringods

**Review by ringods — APPROVED** [review]

**Review by guineveresaenger — CHANGES_REQUESTED** [review]
There seems to be a file referenced that doesn't exist (yet)? Please disregard if GitHub is playing tricks on me.

Can we consider consolidating https://github.com/pulumi/pulumi-terraform-bridge/blob/master/pkg/tfbridge/README.md to here perhaps?

File: docs/metadata.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Provider Metadata
+
+Bridged providers may need to persist information between different invocations of `make
+tfgen`, or between tfgen time and normal runtime. The bridge provides a mechanism to do
+that: metadata.
+
+Provider metadata is required for PF based providers, muxed providers and [AutoAliasing](./auto-aliasing.md).
```
[the remark this answers is not in the document]
Comment (iwahbe) [reply docs/metadata.md:7]: Fixed. I meant to move that section of the readme, but forgot.

**Review by guineveresaenger — APPROVED** [review]
LGTM!

**Pull request merged**